osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -0,0 +1,187 @@
+const testProject = "compute-image-test-pool-001"
+const testZone = "us-central1-c"
+
+var dump = &pretty.Config{IncludeUnexported: true}
+
+// TODO: Move to the new combined osconfig package, also make this easily available to other tests.
+var installOsConfigAgentDeb = `echo 'deb http://packages.cloud.google.com/apt google-osconfig-agent-stretch-unstable main' >> /etc/apt/sources.list
+curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+apt-get update
+apt-get install -y google-osconfig-agent
+echo 'osconfig agent install done'`
+var installOsConfigAgentGooGet = `c:\programdata\googet\googet.exe -noconfirm install -sources https://packages.cloud.google.com/yuck/repos/google-osconfig-agent-unstable google-osconfig-agent
+echo 'osconfig agent install done'`
+var installOsConfigAgentYumEL7 = `cat > /etc/yum.repos.d/google-osconfig-agent.repo <<EOM
+[google-osconfig-agent]
+name=Google OSConfig Agent Repository
+baseurl=https://packages.cloud.google.com/yum/repos/google-osconfig-agent-el7-unstable
+enabled=1
+gpgcheck=0
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+       https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOM
+yum -y install google-osconfig-agent
+echo 'osconfig agent install done'`
+
+type packageManagementTestSetup struct {
+	image       string
+	name        string
+	packageType []string
+	shortName   string
+	startup     *api.MetadataItems
+}
+
+// TestSuite is a PackageManagementTests test suite.
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	testSetup := []*packageManagementTestSetup{
+		// Debian images.
+		&packageManagementTestSetup{
+			image:       "projects/debian-cloud/global/images/family/debian-9",
+			packageType: []string{"deb"},
+			shortName:   "debian",
+			startup: &api.MetadataItems{
+				Key:   "startup-script",
+				Value: &installOsConfigAgentDeb,
+			},
+		},
+	}
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range testSetup {
+		wg.Add(1)
+		go packageManagementTestCase(ctx, setup, tests, &wg, logger, testCaseRegex)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger) {
+	defer cleanupOsConfig(ctx, testCase, logger)
+
+	req := &osconfigpb.CreateOsConfigRequest{
+		Parent: "projects/compute-image-test-pool-001",
+		OsConfig: &osconfigpb.OsConfig{
+			Name: "foo",
```

Comment:
consider "create-osconfig-test-osconfig" or something with a little more meaning to help debug if you're looking at the project.

Reply:
changing the project will require some setup in the new project(test-infra, prow, etc). Also, I was talking to Liam today about our efforts on making our tools repo CI/CD. We have plans to redo our setup in order to move towards CI/CD pipeline.
I would suggest, for now lets keep the tests in this project for temporary period and once we get more idea about how our CI/CD   pipeline would look like, we can make the changes.

Reply:
To clarify, I was referring to the `Name` of the OsConfig and not the project where it resides. I'm fine with whatever project(s) we decide on. But if someone stumbles into the project via the console or gcloud, giving these resources identifying names will be helpful.

Reply:
ah..ok.. yeah sure. will change the name of the osconfig.